teamcity: failed tests on master: testrace/TestRegistryResumeExpiredLease

The following tests appear to have failed:

[#342179](https://teamcity.cockroachdb.com/viewLog.html?buildId=342179):

```
--- FAIL: testrace/TestRegistryResumeExpiredLease (0.000s)
Race detected!

------- Stdout: -------
I170902 00:55:41.842630 4362 server/config.go:520  [n?] 1 storage engine initialized
I170902 00:55:41.842779 4362 server/config.go:522  [n?] RocksDB cache size: 128 MiB
I170902 00:55:41.842905 4362 server/config.go:522  [n?] store 0: in-memory, size 100 MiB
I170902 00:55:41.865804 4362 server/node.go:344  [n?] **** cluster 8d430b1f-97d6-4c9b-891d-a59243930f89 has been created
I170902 00:55:41.865873 4362 server/server.go:833  [n?] **** add additional nodes by specifying --join=127.0.0.1:38023
I170902 00:55:41.874436 4362 storage/store.go:1193  [n1,s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170902 00:55:41.874747 4362 server/node.go:461  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=3.2 KiB), ranges=1, leases=0, writes=0.00, bytesPerReplica={p10=3266.00 p25=3266.00 p50=3266.00 p75=3266.00 p90=3266.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00}
I170902 00:55:41.874901 4362 server/node.go:326  [n1] node ID 1 initialized
I170902 00:55:41.875157 4362 gossip/gossip.go:311  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:38023" > attrs:<> locality:<> ServerVersion:<major_val:1 minor_val:0 patch:0 unstable:3 > 
I170902 00:55:41.875932 4362 storage/stores.go:303  [n1] read 0 node addresses from persistent storage
I170902 00:55:41.876202 4362 server/node.go:606  [n1] connecting to gossip network to verify cluster ID...
I170902 00:55:41.876721 4362 server/node.go:631  [n1] node connected via gossip and verified as part of cluster "8d430b1f-97d6-4c9b-891d-a59243930f89"
I170902 00:55:41.877077 4362 server/node.go:403  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I170902 00:55:41.940145 4362 sql/executor.go:402  [n1] creating distSQLPlanner with address {tcp 127.0.0.1:38023}
I170902 00:55:41.985570 4517 storage/replica_command.go:2710  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I170902 00:55:42.014800 4362 server/server.go:925  [n1] starting https server at 127.0.0.1:45343
I170902 00:55:42.014887 4362 server/server.go:926  [n1] starting grpc/postgres server at 127.0.0.1:38023
I170902 00:55:42.014932 4362 server/server.go:927  [n1] advertising CockroachDB node at 127.0.0.1:38023
W170902 00:55:42.015225 4362 sql/jobs/registry.go:156  [n1] unable to get node liveness: node not in the liveness table
E170902 00:55:42.239017 4412 storage/consistency_queue.go:107  [replica consistency checker,n1,s1,r1/1:/{Min-System/}] key range /Min-/Max outside of bounds of range /Min-/System/""
I170902 00:55:42.241461 4517 storage/replica_command.go:2710  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
I170902 00:55:42.284305 4362 sql/event_log.go:102  [n1] Event: "alter_table", target: 12, info: {TableName:eventlog Statement:ALTER TABLE system.eventlog ALTER COLUMN "uniqueID" SET DEFAULT uuid_v4() User:node MutationID:0 CascadeDroppedViews:[]}
I170902 00:55:42.306355 4517 storage/replica_command.go:2710  [split,n1,s1,r3/1:/{System/NodeL���-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I170902 00:55:42.347602 4362 sql/lease.go:341  [n1] publish: descID=12 (eventlog) version=2 mtime=2017-09-02 00:55:42.34404513 +0000 UTC
I170902 00:55:42.380640 4517 storage/replica_command.go:2710  [split,n1,s1,r4/1:/{System/NodeL���-Max}] initiating a split of this range at key /System/tsd [r5]
I170902 00:55:42.502979 4517 storage/replica_command.go:2710  [split,n1,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
I170902 00:55:42.573183 4517 storage/replica_command.go:2710  [split,n1,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
I170902 00:55:42.622224 4517 storage/replica_command.go:2710  [split,n1,s1,r7/1:/{Table/System���-Max}] initiating a split of this range at key /Table/11 [r8]
I170902 00:55:42.665354 4517 storage/replica_command.go:2710  [split,n1,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
I170902 00:55:42.676727 4362 sql/event_log.go:102  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:node}
I170902 00:55:42.755955 4517 storage/replica_command.go:2710  [split,n1,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I170902 00:55:42.847348 4517 storage/replica_command.go:2710  [split,n1,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I170902 00:55:42.914068 4517 storage/replica_command.go:2710  [split,n1,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I170902 00:55:42.944321 4362 sql/event_log.go:102  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:1.0-3 User:node}
I170902 00:55:43.027822 4517 storage/replica_command.go:2710  [split,n1,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I170902 00:55:43.043313 4362 sql/event_log.go:102  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:node}
I170902 00:55:43.069754 4362 server/server.go:1069  [n1] done ensuring all necessary migrations have run
I170902 00:55:43.070096 4362 server/server.go:1071  [n1] serving sql connections
I170902 00:55:43.070644 4517 storage/replica_command.go:2710  [split,n1,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I170902 00:55:43.092237 5060 sql/event_log.go:102  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:38023} Attrs: Locality: ServerVersion:1.0-3} ClusterID:8d430b1f-97d6-4c9b-891d-a59243930f89 StartedAt:1504313741876761700 LastUp:1504313741876761700}
I170902 00:55:43.137140 4517 storage/replica_command.go:2710  [split,n1,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
I170902 00:55:43.159809 4517 storage/replica_command.go:2710  [split,n1,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I170902 00:55:43.784553 5531 sql/jobs/registry.go:293  job 276048587269832705: resuming
I170902 00:55:43.939154 5650 sql/jobs/registry.go:293  job 276048588094799873: resuming
I170902 00:55:43.940916 4362 util/stop/stopper.go:476  quiescing; tasks left:
1      [async] kv.TxnCoordSender: heartbeat loop
E170902 00:55:43.941390 5650 sql/jobs/jobs.go:377  ignoring error while marking job 276048588094799873 as successful: failed to send RPC: sending to all 1 replicas failed; last error: {(err: node unavailable; try another peer) <nil>}
I170902 00:55:43.942640 4519 kv/transport_race.go:69  transport race promotion: ran 22 iterations on up to 798 requests
==================
Read at 0x00c420f2e0bc by goroutine 56:
  internal/race.Read()
      /usr/local/go/src/internal/race/race.go:37 +0x38
  sync.(*WaitGroup).Add()
      /usr/local/go/src/sync/waitgroup.go:71 +0x26b
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:190 +0x84
  github.com/cockroachdb/cockroach/pkg/kv.GRPCTransportFactory()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport_race.go:106 +0x33f
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendToReplicas()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:1123 +0x3a7
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendRPC()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:382 +0x352
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendSingleRange()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:446 +0x1b7
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendPartialBatch()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:930 +0x574
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).divideAndSendBatchToRanges()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:795 +0xd1d
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:597 +0x3ed
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:444 +0x281
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).send()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:545 +0x209
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:908 +0x577
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Send-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:486 +0xa3
  github.com/cockroachdb/cockroach/pkg/internal/client.sendAndFill()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:454 +0x155
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:486 +0x111
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Get()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:337 +0x10a
  github.com/cockroachdb/cockroach/pkg/sql.getDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/descriptor.go:172 +0xdc
  github.com/cockroachdb/cockroach/pkg/sql.getDatabaseDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/database.go:125 +0x173
  github.com/cockroachdb/cockroach/pkg/sql.MustGetDatabaseDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/database.go:151 +0x93
  github.com/cockroachdb/cockroach/pkg/sql.getTableOrViewDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:111 +0x19a
  github.com/cockroachdb/cockroach/pkg/sql.getTableDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:136 +0x80
  github.com/cockroachdb/cockroach/pkg/sql.MustGetTableDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:191 +0x91
  github.com/cockroachdb/cockroach/pkg/sql.(*TableCollection).getTableVersion()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:307 +0x17e
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).makeEditNode()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/update.go:51 +0x107
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).Update()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/update.go:219 +0x156
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).newPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:451 +0x164f
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).makePlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:187 +0xa2
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).query()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planner.go:233 +0x1c0
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planner.go:286 +0xbe
  github.com/cockroachdb/cockroach/pkg/sql.InternalExecutor.ExecuteStatementInTransaction()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:47 +0x15f
  github.com/cockroachdb/cockroach/pkg/sql/jobs.(*Job).update.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/jobs.go:538 +0x71c
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:507 +0x5e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:722 +0xc5
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:508 +0x174
  github.com/cockroachdb/cockroach/pkg/sql/jobs.(*Job).runInTxn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/jobs.go:436 +0x208
  github.com/cockroachdb/cockroach/pkg/sql/jobs.(*Job).update()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/jobs.go:546 +0x1a4
  github.com/cockroachdb/cockroach/pkg/sql/jobs.(*Job).Succeeded()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/jobs.go:325 +0xd8
  github.com/cockroachdb/cockroach/pkg/sql/jobs.(*Job).FinishedWith()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/jobs.go:374 +0x25a
  github.com/cockroachdb/cockroach/pkg/sql/jobs.(*Registry).maybeAdoptJob.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/registry.go:313 +0x1bb

Previous write at 0x00c420f2e0bc by goroutine 84:
  internal/race.Write()
      /usr/local/go/src/internal/race/race.go:41 +0x38
  sync.(*WaitGroup).Wait()
      /usr/local/go/src/sync/waitgroup.go:129 +0x14b
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:423 +0x2d2
  github.com/cockroachdb/cockroach/pkg/sql/jobs_test.TestRegistryResumeExpiredLease()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/registry_external_test.go:173 +0x128e
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:657 +0x107

Goroutine 56 (running) created at:
  github.com/cockroachdb/cockroach/pkg/sql/jobs.(*Registry).maybeAdoptJob()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/registry.go:317 +0xbdb
  github.com/cockroachdb/cockroach/pkg/sql/jobs.(*Registry).Start.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/registry.go:142 +0x154
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:198 +0x160

Goroutine 84 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:697 +0x543
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:882 +0xaa
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:657 +0x107
  testing.runTests()
      /usr/local/go/src/testing/testing.go:888 +0x4e0
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:822 +0x1c3
  github.com/cockroachdb/cockroach/pkg/sql/jobs_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/main_test.go:33 +0x189
  main.main()
      github.com/cockroachdb/cockroach/pkg/sql/jobs/_test/_testmain.go:54 +0x20f
==================
I170902 00:55:34.845565 1 rand.go:74  Random seed: -2435538140190066123

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/kv/transport_race.go
+++ b/pkg/kv/transport_race.go
@@ -57,7 +57,9 @@ func GRPCTransportFactory(
 	opts SendOptions, rpcContext *rpc.Context, replicas ReplicaSlice, args roachpb.BatchRequest,
 ) (Transport, error) {
 	if atomic.AddInt32(&running, 1) <= 1 {
-		rpcContext.Stopper.RunWorker(context.TODO(), func(ctx context.Context) {
+		// NB: We can't use Stopper.RunWorker because doing so would race with
+		// calling Stopper.Stop.
+		rpcContext.Stopper.RunAsyncTask(context.TODO(), "transport racer", func(ctx context.Context) {
 			var iters int
 			var curIdx int
 			defer func() {
@@ -92,7 +94,7 @@ func GRPCTransportFactory(
 				// then access everything we have.
 				for {
 					select {
-					case <-rpcContext.Stopper.ShouldStop():
+					case <-rpcContext.Stopper.ShouldQuiesce():
 						return
 					case ba := <-incoming:
 						bas[curIdx%size] = ba

